emulator/src/stressors/cpu.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2023 Telefonaktiebolaget LM Ericsson AB
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stressors
+
+import (
+	"application-emulator/src/util"
+	model "application-model"
+	"fmt"
+	"runtime"
+	"sync"
+)
+
+type CPUTask struct{}
+
+// Combines the CPU task response in taskResponses with cpuTaskResponse
+func ConcatenateCPUResponses(taskResponses *MutexTaskResponses, cpuTaskResponse *model.CPUTaskResponse) {
+	taskResponses.Mutex.Lock()
+	defer taskResponses.Mutex.Unlock()
+
+	if taskResponses.CPUTask != nil {
+		taskResponses.CPUTask.Services = append(taskResponses.CPUTask.Services, cpuTaskResponse.Services...)
+		taskResponses.CPUTask.Statuses = append(taskResponses.CPUTask.Statuses, cpuTaskResponse.Statuses...)
+	} else {
+		taskResponses.CPUTask = cpuTaskResponse
+	}
+}
+
+func (c *CPUTask) ExecAllowed(endpoint *model.Endpoint) bool {
+	return endpoint.CpuComplexity != nil
+}
+
+func StressCPU(executionTime float32, lockThread bool) {
+	// TODO: This needs to be tested more
+	if executionTime > 0 {
+		// TODO: Threads need to be locked because otherwise util.ThreadCPUTime() can change in the middle of execution
```

Comment:
This seems already implemented, should we keep it as TODO?

Reply:
I'm not sure if it's the right approach (since Go programs normally don't use `LockOSThread`), so I think we should keep the TODO for now. I'm going to clean up all the TODOs at the end when all functionality is implemented.